format/materialized-view-spec.md

```diff
@@ -0,0 +1,131 @@
+Iceberg Materialized views are a way to store the results of an Iceberg view to reuse the computation in subsequent queries.
+
+## Goals 
+The goal of this spec is to define the metadata associated with materialized views in Iceberg.
+Such metadata allows creating and querying Iceberg materialized views across different engines.
+
+## Specification
+A materialized view is an Iceberg view with a respective Iceberg table that stores the results of the view query.
+An Iceberg view is considered a materialized view if it has the `materialized.view` property set to `true`.
+A materialized view must also reference the storage table identifier in its `materialized.view.storage.table` property.
+
+The specification for the materialized view properties on the view is as follows:
+| Property name                              | Description                                                             |
+|--------------------------------------------|-------------------------------------------------------------------------|
+| `materialized.view`                | This property is used to mark whether a view is a materialized view. If set to `true`, the view is treated as a materialized view.|
+| `materialized.view.storage.table`  | This property specifies the identifier of the storage table associated with the materialized view.|
+
+In addition to the properties on the view, the storage table associated with the materialized view has the following properties:
+
+| Property name                        | Description                                                                   |
+|--------------------------------------|-------------------------------------------------------------------------------|
+| `base.snapshot.[UUID]`       | These properties store the snapshot IDs of the base tables at the time the materialized view's data was last updated. Each property is prefixed with `base.snapshot.` followed by the UUID of the base table.|
+| `view.version`               | This property stores the version of the view that this storage table is associated with at the time of materialization|
+| `child.view.version.[UUID]`  | If the top level materialized view is nested on top of other views, these properties store the version of the child views that the top level view is associated with at the time of materialization. Each property is prefixed with `child.view.version.` followed by the UUID of the child view.|
+
+A storage table is considered fresh if all the following conditions are true:
+* For each table `T` with `UUID` in `base.snapshot.[UUID]`, the current snapshot ID of `T` is equal to the value stored in `base.snapshot.[UUID]`.
+* The version of the view using the table as the storage table is equal to the value stored in `view.version`.
+* For each child view `V` with `UUID` in `child.view.version.[UUID]`, the version of `V` is equal to the value stored in `child.view.version.[UUID]`.
```

Comment:
there is a chance that storage table of child views are outdated (not fresh). So, just comparing the view versions is not enough to conclude child views are fresh?

Reply:
In case child views are materialized, their storage table is not relevant to determining the freshness of the current (top level view) storage table. 

> is not enough to conclude child views are fresh
The objective is not to tell whether child views are fresh, but rather if the current storage table is a valid materialization of the current view tree. If the view tree changes in any way, the current table is no longer a valid materialization for the current view tree.

Reply:
I have added the above explanation to the spec text.